src/lhttpc_sock.erl

```diff
@@ -68,7 +68,15 @@
 -spec connect(host(), integer(), socket_options(), timeout(), boolean()) ->
     {ok, socket()} | {error, atom()}.
 connect(Host, Port, Options, Timeout, true) ->
-    ssl:connect(Host, Port, Options, Timeout);
+    %% Server Name Indication is this:
+    %% https://en.wikipedia.org/wiki/Server_Name_Indication
+    %%
+    %% We inject this option here because as of OTP 20, the server name is
+    %% passed back as a return value so it can be validated by the client.
+    %% We want the previous behavior which ignored the server name
+    %% indicator, so disable this.  I believe this option has existed
+    %% since R16.
+    ssl:connect(Host, Port, Options ++ [{server_name_indication, disable}], Timeout);
```

Comment:
should this be `[{server_name_indication, disable} | Options ]`? Meaning, is any other value basically incompatible with lhttpc?

Reply:
Probably would be better than what I have.

Reply:
it should be first param anyway to make proplists:get work nicely if smb has actually passed smth

Reply:
OK, I will change. 1 sec